### ISSUE TITLE
Problem on reproducting logits-based method about MLP training

### ISSUE THREAD

**Issue by FireCaramelPudding** [TsinghuaC3I/CoGenesis#2]
Hello, thanks for sharing this great work! I encountered an issue during MLP training reproduction about logits-based method:

When the ground truth token is outside the TOP-K predictions of both the small model and large model, the model struggles to learn proper weight allocation between them. This leads to persistently high loss values during training.

Have you encountered this situation? Could you please share:

How this edge case was handled in your implementation?
Any recommended strategies to improve learning in such scenarios?

Additional context:

My reproduction environment:
```
vllm==0.5.1 ## have Modified the logprob generation
transformers==4.47.0
pytorch==2.3.0
```
Datasets: follow `ctx_train` and `ctx_dev`
Observed behavior: Loss stays at about 20 after 3 epochs iterations when this occurs, and only few tokens(about 5?) in a sequence are in small model or large model TOP-K predictions

**Comment by iseesaw** [comment]
Sorry for the late response. Our org’s issues don’t send automatic email notifications, so we missed this. We didn’t see this on our side. Two quick things that usually fix it: (1) always force-include the gold token (with its original logit) in each model’s candidate set; (2) avoid -1e9 padding, mix after log_softmax or clamp to a milder floor (≈−20). If that still doesn’t help, share the % of tokens where the gold is outside both sets and we can take another look.
